packages/lib/src/utils/ContractAST.ts

```diff
@@ -103,23 +115,52 @@ export default class ContractAST {
       });
   }
 
+  // This method is used instead of getLinearizedBaseContracts only because
+  // it keeps track of the names as well as the IDs of the ancestor contracts,
+  // and can yield a better error message to the user than "AST node NN not found"
+  public getBaseContractsRecursively(): Node[] {
+    const mapBaseContracts = (baseContracts) => (baseContracts.map(c => ({
+      id: c.baseName.referencedDeclaration, name: c.baseName.name
+    })));
+    const baseContractsToVisit = mapBaseContracts(this.getContractNode().baseContracts);
+    const visitedBaseContracts = {};
+
+    while (baseContractsToVisit.length > 0) {
+      const { id, name } = baseContractsToVisit.pop();
+      if (visitedBaseContracts[id]) continue;
+
+      try {
+        const node = this.getNode(id, 'ContractDefinition');
+        visitedBaseContracts[id] = node;
+        baseContractsToVisit.push(... mapBaseContracts(node.baseContracts));
+      } catch (err) {
+        if (err instanceof NodeNotFoundError) {
+          throw new Error(`Cannot find source data for contract ${name} (base contract of ${this.contract.schema.contractName}). This often happens because either:\n- An incremental compilation step went wrong. Clear your build folder and recompile.\n- There is more than one contract named ${name} in your project (including dependencies). Make sure all contracts have a unique name, and that you are not importing dependencies with duplicated contract names (for example, openzeppelin-eth and openzeppelin-solidity).`);
```

Comment:
I was on the fence on whether to go with something this long or not. However, since it's a fairly uncommon scenario, I opted for a more thorough description.